extras/beast/unit_test/suite.hpp

```diff
@@ -44,95 +45,100 @@ class suite
         char const*
         what() const noexcept override
         {
-            return "suite aborted";
+            return "test suite aborted";
         }
     };
 
-public:
-    // Memberspace
-    class log_t
+    template<class CharT, class Traits, class Allocator>
+    class log_buf
+        : public std::basic_stringbuf<CharT, Traits, Allocator>
     {
-    private:
-        friend class suite;
-        suite* suite_ = nullptr;
+        suite& suite_;
 
     public:
-        log_t () = default;
+        explicit
+        log_buf(suite& self)
+            : suite_(self)
+        {
+        }
 
-        template <class T>
-        abstract_ostream::scoped_stream_type
-        operator<< (T const& t);
+        ~log_buf()
+        {
+            sync();
+        }
 
-        /** Returns the raw stream used for output. */
-        abstract_ostream&
-        stream();
+        int
+        sync() override
+        {
+            auto const& s = this->str();
```

Comment:
@nbougalis I had the same question. gcc give a nice explanation (along with a fix!):

`extras/beast/unit_test/suite.hpp:76:16: error: ‘str’ was not declared in this scope, and no declarations were found by argument-dependent lookup at the point of instantiation [-fpermissive]
             str("");
                ^
extras/beast/unit_test/suite.hpp:76:16: note: declarations in dependent base ‘std::__cxx11::basic_stringbuf<char>’ are not found by unqualified lookup
extras/beast/unit_test/suite.hpp:76:16: note: use ‘this->str’ instead`